### ISSUE TITLE
convert MP4 to MP4 (to compress the file)

### ISSUE THREAD

**Issue by jakobroehrl** [PaulLereverend/NextcloudVideo_Converter#56]
I would like to convert an MP4 file to an compressed MP4 file. 
Currently it's only possible to choose a export filetype that is unequal to the input filetype

**Comment by UniqueActive** [comment]
I was looking for that feature too, for two reasons:

1. Being able to compress an file to the same format inside of Nextcloud means I don't have to download an unnecessarily big video and compress it locally, only to reupload it
2. An MP4 that is not streamable through Nextcloud or its app can be converted to a streamable one

This feature was my primary motivation for installing the video converter, so I would be very happy to see its inclusion, as it seems highly useful to me and I don't see any reasons not to allow it :)

**Comment by BugerDread** [comment]
Making files just smaller (compress them more) was also my primary motivation to install this plugin, simply because mp4 videos from my phone are so big that it is impossible to stream them fluently in nextcloud over the internet (because 52 seconds of video use 95MB and my upload is not so fast).

Tried to convert them (using your plugin) to .avi and is seems that nextcloud refuse to stream it (im offered to download the file when clicked), **it seems that it works when converted to m4v** so Im happy anyway - thank you.

**Comment by veale** [comment]
Yes, I was surprised you couldn't do this too. Is it an issue with overwriting the same file? Even if I do it but add some suffix like video-compressed.mp4 it would be useful. I understand that overwriting the file automatically might be problematic in case it was lost due to an error.

**Comment by Bellusterra** [comment]
To add to this, having a setting in the admin area to change the CRF conversion quality as well as other such settings if any would be great to give the admin flexibility based on their and their users bandwidth availability.

**Comment by Bellusterra** [comment]
Was playing around with PiGallery2 and implementing something similar to that would be good I think. You choose the video size, crf, frame rate and any custom commands and it will generate video files that are streamed when playing online, downloading that video pulls the original.

**Comment by tourtools** [comment]
rename the file to [filename].MP4 and run the conversion, the control responsible of disabling the button is case sensitive

**Comment by BugerDread** [comment]
> rename the file to [filename].MP4 and run the conversion, the script responsible of disabling the button is case sensitive

LOL, what a nice hack :D
IT WORKS :D
